src/librustc_typeck/check/_match.rs

```diff
@@ -791,6 +792,22 @@ https://doc.rust-lang.org/reference/types.html#trait-objects");
 
         // Resolve the path and check the definition for errors.
         let (def, opt_ty, segments) = self.resolve_ty_and_def_ufcs(qpath, pat.id, pat.span);
+        if def == Def::Err {
+            return self.tcx.types.err;
```

Comment:
I think you need to perform some additional actions for `Def::Err` - call `set_tainted_by_errors`, `on_error`.
There's code below doing it.

Reply:
Added.